### ISSUE TITLE
Documented issues from troubleshooting on Windows

### ISSUE THREAD

**Issue by Kim2091** [lkwq007/stablediffusion-infinity#12]
I helped 4 separate people in the last 5 hours set this software up on Windows, here's what issues I've run into and the solutions. I apologize for this being condensed into a single issue, but I'm low on time:

Problem 1: **Tokens**
Many people have been having issues using `huggingface cli login` to register their token for usage with this software. The workaround to this is to use `use_auth_token='tokenvalue'` (where tokenvalue is the token) rather than `use_auth_token=True`. This shouldn't be the default, but it should be documented as a potential workaround.

Problem 2: **AttributeError: module 'PIL.Image' has no attribute 'Resampling'**
This issue only occurred for one person. No amount of re-running existing cells fixes it, and all dependencies are installed. 

Problem 3: **OSError: Windows requires Developer Mode to be activated, or to run Python as an administrator, in order to create symlinks.**
This doesn't seem to be an issue with this software as much as it is with permissions in general, but it's still an issue that's odd. It didn't occur for 3 out of the 4 people I was helping. The person who ran into this issue had an administrator account, and was running it under the same conditions as the other users. Developer mode did fix it.

Problem 4: **OpenCV**
Installing `opencv` with conda seems to prevent errors with patch_match. This is the command I've been recommending and it has successfully prevented it: `conda install -c conda-forge opencv`.

Concerns:
**No such comm target registered: jupyter.widget.control**
Full error: 
> [IPKernelApp] ERROR | No such comm target registered: jupyter.widget.control
> [IPKernelApp] WARNING | No such comm: a5f38cb6-3d6d-46b5-8493-fea3b4bc45fb

There's no clear reason as to why this appears, but it shows up in the log. Functionality does not appear to be inhibited.

**Comment by Kim2091** [comment]
3/4 people I helped did *not* require developer mode, which is quite odd. (I didn't require it either)

**Comment by Vasissualiyp** [comment]
After following all of the instructions, I still get the error:

```

> ---------------------------------------------------------------------------
> ImportError                               Traceback (most recent call last)
> Cell In [1], line 6
>       3 from ipycanvas import Canvas, hold_canvas, MultiCanvas
>       4 import ipywidgets as widgets
> ----> 6 from utils import *
>       7 from canvas import InfCanvas
> 
> File ~\stablediffusion-infinity\utils.py:3
>       1 from PIL import Image
>       2 from PIL import ImageFilter
> ----> 3 import cv2
>       4 import numpy as np
>       5 import scipy
> 
> ImportError: DLL load failed while importing cv2: The specified module could not be found.
```

That also results in the error in the next cell:

```
ImportError                               Traceback (most recent call last)
Cell In [6], line 4
      2 import torch
      3 from torch import autocast
----> 4 from diffusers import StableDiffusionPipeline, StableDiffusionInpaintPipeline

File ~\Anaconda3\envs\sd-inf\lib\site-packages\diffusers\__init__.py:12
      1 from .utils import (
      2     is_inflect_available,
      3     is_onnx_available,
   (...)
      6     is_unidecode_available,
      7 )
     10 __version__ = "0.3.0"
---> 12 from .configuration_utils import ConfigMixin
     13 from .modeling_utils import ModelMixin
     14 from .models import AutoencoderKL, UNet2DConditionModel, UNet2DModel, VQModel

File ~\Anaconda3\envs\sd-inf\lib\site-packages\diffusers\configuration_utils.py:25
     22 from collections import OrderedDict
     23 from typing import Any, Dict, Tuple, Union
---> 25 from huggingface_hub import hf_hub_download
     26 from huggingface_hub.utils import EntryNotFoundError, RepositoryNotFoundError, RevisionNotFoundError
     27 from requests import HTTPError

ImportError: cannot import name 'hf_hub_download' from 'huggingface_hub' (C:\Users\vaspu\Anaconda3\envs\sd-inf\lib\site-packages\huggingface_hub\__init__.py)
```

**Comment by joeybab3** [comment]
For windows even after reinstalling using only the conda command for opencv I'm still getting `patch_match compiling failed
![image](https://user-images.githubusercontent.com/2829457/190837111-26a2f7ca-b2f6-4471-a4e2-139d0524cf1b.png)
`

**Comment by Kim2091** [comment]
Sadly I'm not sure what's causing your guys' issues here. I had a fairly small sample set of users and I myself didn't run into these (that weren't fixed by my solutions posted above).

**Comment by joeybab3** [comment]
Well luckily it worked despite the error message...

**Comment by orangpelupa** [comment]
`workaround to this is to use use_auth_token='tokenvalue'`

where do i put  use_auth_token='tokenvalue'?

i put it in huggingface-cli launch parameter and it says incorrect parameter

**Comment by orangpelupa** [comment]
got this error on windows. ive tried pip install infcanvas and theres no infcanvas

```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In [1], line 7
      4 import ipywidgets as widgets
      6 from utils import *
----> 7 from canvas import InfCanvas

ImportError: cannot import name 'InfCanvas' from 'canvas' (B:\tools\anaconda3\envs\sd-inf\lib\site-packages\canvas\__init__.py)
```

**Comment by Loathesong** [comment]
I have a hunch that the patch_match issue and it only affecting some users is due to some using WSL and others running directly off of windows. 
I believe the library needs to be compiled on linux/WSL

**Comment by Kim2091** [comment]
> I have a hunch that the patch_match issue and it only affecting some users is due to some using WSL and others running directly off of windows.
> I believe the library needs to be compiled on linux/WSL

That's possible. It's worth noting that of myself and the 4 others, none were running WSL.

**Comment by Kallamamran** [comment]
NameError                                 Traceback (most recent call last)
Cell In [5], line 22, in run_outpaint(btn)
     20     # mask_image=mask_image.filter(ImageFilter.GaussianBlur(radius = 8))
     21     with autocast("cuda"):
---> 22         images = inpaint(
     23             prompt=base.text_input.value,
     24             init_image=init_image.resize(
     25                 (process_size, process_size), resample=Image.Resampling.LANCZOS
     26             ),
     27             mask_image=mask_image.resize((process_size, process_size)),
     28             strength=strength_slider.value,
     29             num_inference_steps=step_input.value,
     30             guidance_scale=guidance_input.value,
     31         )["sample"]
     32 else:
     33     with autocast("cuda"):

NameError: name 'inpaint' is not defined

**Comment by CrashAnimus** [comment]
> Problem 1: Tokens
> Many people have been having issues using huggingface cli login to register their token for usage with this software. The workaround to this is to use use_auth_token='tokenvalue' (where tokenvalue is the token) rather than use_auth_token=True. This shouldn't be the default, but it should be documented as a potential workaround.

The solution for me was to not use CTRL+V to paste the token.   If using VScode, use CTRL+SHIFT+V, if using cmd/powershell outside of VScode, just right-click in the terminal to paste.

**Comment by mattrat1988** [comment]
i keep running into this issue when uploading a photo not sure what it means 



AttributeError                            Traceback (most recent call last)
Cell In [21], line 14, in start_func(btn)
     12 uploader_output.clear_output()
     13 if len(uploader.value)>0:
---> 14     keys=list(uploader.value.keys())
     15     val=uploader.value[keys[-1]]["content"]
     16     pil=Image.open(io.BytesIO(val))

AttributeError: 'tuple' object has no attribute 'keys'

**Comment by Crucialhunter** [comment]
I ran everything but when trying to run 
python app.py
y get 
 python: can't open file 'C:\\Windows\\system32\\app.py': [Errno 2] No such file or directory
 
 Anyone can help me understand what im missing? thanks!

**Comment by abulka** [comment]
My installation experience on Windows had some bumps along the way:
- The infinity repo needs to be cloned - that is presumably so obvious that it isn't mentioned in the setup instructions. So remember to `git clone git@github.com:lkwq007/stablediffusion-infinity.git`.
- Pytorch ran off CPU only not GPU (cuda) so I changed the pytorch installation line to `conda install pytorch torchvision torchaudio cudatoolkit -c pytorch -c nvidia -c conda-forge` which fixed it. See https://github.com/lkwq007/stablediffusion-infinity/issues/123#issuecomment-1299401651
- When I hit the setup button in the UI, models wern't downloading from HuggingFace so I generated a token from https://huggingface.co/settings/tokens but didn't know where to put it. Turns out you can paste it into UI into the huggingface area I think, though I put it into a file called `.token` in the root of the project.

Some of this information should potentially be in the setup instructions, to help others.

**Comment by reidprichard** [comment]
I agree that it would be helpful to point out the need to `git clone` the repository - I was about to give up on getting this working until I found this thread.
